ragna/core/_rag.py

```diff
@@ -130,15 +196,13 @@ def __init__(
         self,
         rag: Rag,
         *,
-        documents: Iterable[Any],
```

Comment:
We need to be able to also pass in regular paths here. While `Corpus` is a useful abstraction, it is unnecessary for the base case. I would move the `_parse_documents` function that you moved to `Corpus` back here and let it return a `Corpus` in case we encounter a list of paths.